mne/decoding/tests/test_csp.py

```diff
@@ -103,6 +103,37 @@ def test_csp():
         assert_array_equal(csp.filters_.shape, [n_channels, n_channels])
         assert_array_equal(csp.patterns_.shape, [n_channels, n_channels])
 
+    # Test average power transform
+    # Check default
+    csp = CSP(log=None)
+    assert_true(csp.log is True)
```

Comment:
It's to prevent user to forget about the log param, as @alexandrebarachant's noted.

Specifically, we should ensure that the user doesn't forget about the log param if he/she applies a `csp_space` transform and not an `average_power`. Thus, the logic is
```
log is None by default
if transform_into == 'csp_space':
     assert_true(log is None)
else:
      log = True if log is None else log
     assert_true(isinstance(log, bool))
```

But you're right that I shouldn't change the log value in the init. Shall I move this logic to `self.transform()` ?